src/newTest/java/org/opensearch/test/framework/cluster/LocalCluster.java

```diff
@@ -0,0 +1,373 @@
+    }
+
+    protected static final AtomicLong num = new AtomicLong();
+
+    private final List<Class<? extends Plugin>> plugins;
+    private final ClusterConfiguration clusterConfiguration;
+    private final TestSecurityConfig testSecurityConfig;
+    private Settings nodeOverride;
+    private final String clusterName;
+    private final MinimumSecuritySettingsSupplierFactory minimumOpenSearchSettingsSupplierFactory;
+    private final TestCertificates testCertificates;
+    private final List<LocalCluster> clusterDependencies;
+    private final Map<String, LocalCluster> remotes;
+    private volatile LocalOpenSearchCluster localOpenSearchCluster;
+	private final List<TestIndex> testIndices;
+
+    private LocalCluster(String clusterName, TestSecurityConfig testSgConfig, Settings nodeOverride,
+            ClusterConfiguration clusterConfiguration, List<Class<? extends Plugin>> plugins, TestCertificates testCertificates,
+            List<LocalCluster> clusterDependencies, Map<String, LocalCluster> remotes, List<TestIndex> testIndices) {
+        this.plugins = plugins;
+        this.testCertificates = testCertificates;
+        this.clusterConfiguration = clusterConfiguration;
+        this.testSecurityConfig = testSgConfig;
+        this.nodeOverride = nodeOverride;
+        this.clusterName = clusterName;
+        this.minimumOpenSearchSettingsSupplierFactory = new MinimumSecuritySettingsSupplierFactory(testCertificates);
+        this.remotes = remotes;
+        this.clusterDependencies = clusterDependencies;
+        this.testIndices = testIndices;
+    }
+
+    @Override
+    public void before() throws Throwable {
+        if (localOpenSearchCluster == null) {
+            for (LocalCluster dependency : clusterDependencies) {
+                if (!dependency.isStarted()) {
+                    dependency.before();
+                }
+            }
+
+            for (Map.Entry<String, LocalCluster> entry : remotes.entrySet()) {
+                @SuppressWarnings("resource")
+                InetSocketAddress transportAddress = entry.getValue().localOpenSearchCluster.masterNode().getTransportAddress();
+                nodeOverride = Settings.builder().put(nodeOverride)
+                        .putList("cluster.remote." + entry.getKey() + ".seeds", transportAddress.getHostString() + ":" + transportAddress.getPort())
+                        .build();
+            }
+
+            start();
+        }
+    }
+
+    @Override
+    protected void after() {
+        if (localOpenSearchCluster != null && localOpenSearchCluster.isStarted()) {
+            try {
+                Thread.sleep(1234);
+                localOpenSearchCluster.destroy();
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            } finally {
+                localOpenSearchCluster = null;
+            }
+        }
+    }
+
+    @Override
+    public void close() {
```

Comment:
In some cases, it can be useful to create the `LocalCluster` not as a resource in the test class, but in the test method. Thus, the `LocalCluster` implements `AutoCloseable`. However, right now this is not used. Thus, we could also remove it.

Reply:
Note; we can keep this method to support AutoClostable, but then maybe we would have `stop` call close to reduce the duplicate logic

Reply:
done